engine/swarm/secrets.md

```diff
@@ -0,0 +1,339 @@
+the same high availability guarantees for secrets as for the rest of the swarm
+management data.
+
+>**Warning**: RAFT data is encrypted in Docker 1.13 and higher. If any of your
+Swarm managers run an earlier version, and one of those managers becomes the
+manager of the swarm, the secrets will be stored unencrypted in that node's RAFT
+logs. Before adding any secrets, update all of your manager nodes to Docker 1.13
+to prevent secrets from being written to plain-text RAFT logs.
+
+When you grant a newly-created or running service access to a secret, the
+decrypted secret is mounted into the container in an in-memory filesystem at
+`/run/secrets/<secret_name>`. You can update a service to grant it access to
+additional secrets or revoke its access to a given secret at any time.
+
+A node only has access to (encrypted) secrets if it is a swarm manager or if it
+is running service tasks which have been granted access to the secret. When a
+container task stops running, the decrypted secrets shared to it are unmounted
+from the in-memory filesystem for that container and flushed from the node's
+memory.
+
+If a node loses connectivity to the swarm while it is running a task container
+with access to a secret, the task container still has access to its secrets, but
+cannot receive updates until the node reconnects to the swarm.
+
+You can add or inspect an individual secret at any time, or list all
+secrets. You cannot remove a secret that a running service is
+using. See [Rotate a secret](secrets.md#example-rotate-a-secret) for a way to
+remove a secret without disrupting running services.
+
+In order to update or roll back secrets more easily, consider adding a version
+number or date to the secret name. This is made easier by the ability to control
+the mount point of the secret within a given container.
+
+## Read more about `docker secret` commands
+
+Use these links to read about specific commands, or continue to the
+[example about using secrets with a service](secrets.md#example-use-secrets-with-a-service).
+
+- [`docker secret create`](../reference/commandline/secret_create.md)
+- [`docker secret inspect`](../reference/commandline/secret_inspect.md)
+- [`docker service ls`](../reference/commandline/secret_ls.md)
+- [`docker secret rm`](../reference/commandline/secret_rm.md)
+- [`--secret`](../reference/commandline/service_create.md#create-a-service-with-secrets) flag for `docker service create`
+- [`--secret-add` and `--secret-rm`](../reference/commandline/service_update.md#adding-and-removing-secrets) flags for `docker service update`
+
+## Example: Get started with secrets
+
+This simple example shows how secrets work in just a few commands. For a
+real-world example, continue to
+[Use secrets with a service](secrets.md#example-use-secrets-with-a-service).
+
+1.  Add a secret to Docker. The `docker secret create` command reads standard
+    input.
+
+    ```bash
+    $ echo "This is a secret" | docker secret create secret_data.txt
```

Comment:
Wonder if we should use `secret_data.txt` as a name here, this may be confusing because either users think this is the name of a local file ("where does the `secret_data.txt` file come from?"), or that it must match the filename under which it can be accessed inside the container (which is later shown that it can be set through `target`).

Perhaps something that more obviously shows it's a custom name, such as `my-little-secret` (without file-extension)